### PR TITLE
Implements task status flow

### PR DESCRIPTION
**Pull request by skv93-coder**
Date: Apr 14 24

Developer Name: @skv93-coder 

---

## Issue Ticket Number

https://github.com/Real-Dev-Squad/website-status/issues/1189
## Description
This PR implements task status flow ACC to #1189 

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
[screen-capture (20).webm](https://github.com/Real-Dev-Squad/website-status/assets/68867418/f0c21744-9323-4091-b88b-92731119ea14)

<details>
<summary>Screenshot 1</summary>

</details>

## Test Coverage
Screenshot from 2024-04-25 19-21-17

<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: __tests__/Unit/Components/Tasks/TaskDetails.test.tsx [hunk 1]
```diff
@@ -529,7 +529,7 @@ describe('Task details Edit mode ', () => {
         });
     });
 
-    test('Should set new status on click at any option', async () => {
+    test.skip('Should set new status on click at any option', async () => {
```
[the remark this answers is not in the document]
Comment (skv93-coder) [reply __tests__/Unit/Components/Tasks/TaskDetails.test.tsx:532]: This test case was written for old code. I have converted that code into component and also added multiple test cases for that component. 
I am going to remove this test case via separate PR

**Review by palakgupta2712 — DISMISSED** [review]

**Comment by skv93-coder** [comment]
> image Can we improve upon the alignment, if possible?

Sure but currently running on tight deadline so creating a new [issue](https://github.com/Real-Dev-Squad/website-status/issues/1195) for the same

**Review by palakgupta2712 — APPROVED** [review]

File: src/constants/constants.ts [hunk 2]
```diff
@@ -38,3 +38,7 @@ export const STATUSES = [
     'completed',
     'done',
 ];
+export const MSG_ON_100_PROGRESS =
+    'Proceeding further will make task progress 100%.';
+export const MSG_ON_0_PROGRESS =
+    'Proceeding further will make task progress 0%.';
```
Comment (mehra-sourav) [inline src/constants/constants.ts:44]: A nitpick:
Maybe you can put both the constants you defined under a single object?
`const PROGRESS_MSG = {
         "0": "msg1",
         "100": "msg2" 
 }
`

**Review by mehra-sourav — APPROVED** [review]

File: __tests__/Unit/Components/Tasks/TaskDetails.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (skv93-coder) [reply __tests__/Unit/Components/Tasks/TaskDetails.test.tsx]: Removed so resolving convo

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**